### ISSUE TITLE
class library .NET 4.8 => .NET 6 - problem

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/wpf#9390]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/t/class-library-NET-48--NET-6---probl/10068649)._

---
[regression] [worked-in:.Net framework 4.8]
Hi,

During migration of our existing project targeting .NET framework 4.8 we encountered starnge error:

MC1000 Unknown build error, 'Could not find type 'System.Web.PreApplicationStartMethodAttribute' in assembly 'C:\Program Files\dotnet\packs\Microsoft.NETCore.App.Ref\6.0.5\ref\net6.0\System.Web.dll'.'

![image.png](https://aka.ms/dc/image?name=B05d31129ea644455b672f732d001a7e9637907199127667224_20220613-142511-image.png&tid=05d31129ea644455b672f732d001a7e9637907199127667224)

Basically we have one big class library targeting net6.0-windows that has been used across multiple different smaller projects.

This class library is compileable as a stanalone porject with no errors, but as soon as we reference it in our WPF project, also targeting same net6.0-windows, we got this error.

Any help appreciated...

---
#### Feedback Bot on 6/14/2022, 11:28 AM: 

(private comment, text removed)
#### Feedback Bot on 10/9/2022, 11:42 AM: 

(private comment, text removed)

---
### Original Solutions
(no solutions)

**Comment by harshit7962** [comment]
Can you provide the minimal repro for the same.

**Comment by bgrainger** [comment]
Looks like a duplicate of this question: https://stackoverflow.com/q/72602061/23633

Based on https://stackoverflow.com/a/76564520/23633 and https://github.com/dotnet/sdk/issues/21942#issuecomment-603193785, it's probably caused by references to old NuGet packages.
